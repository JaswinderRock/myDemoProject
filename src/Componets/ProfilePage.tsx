import React, { Component } from 'react';
import '../css/profile.css';
import ProfileLogo from '../images/profilelogo.png';
import { Link } from 'react-router-dom';

class ProfilePage extends Component {

    render() {
        return (
            <div className='card' style={{ borderRadius: "4px" }}>
                <h2 style={{ paddingTop: '20px', fontSize: '28px' }}> My Profile</h2>
                <img src={ProfileLogo} alt="John" style={{ width: "90%" }} />
                <h1>Miss Jazz</h1>
                <p className="title">Software Engineer</p>
                <p>Reckonsys Tech Labs</p>
                <button><Link to="/dashboard" style={{ color: 'white' }}>Close</Link></button>
            </div>
        )
    }
}
export default ProfilePage;