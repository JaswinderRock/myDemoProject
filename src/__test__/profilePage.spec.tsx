import React from 'react';
import { ShallowWrapper, shallow } from "enzyme";
import ProfilePage from '../Componets/ProfilePage';
import { Link } from 'react-router-dom';

describe("Render Profile Page", () => {
    let wrapper: ShallowWrapper;
    beforeAll(() => {
        wrapper = shallow(<ProfilePage />);
    });
    test("Profile test is render", () => {
        const component = shallow(<ProfilePage />);
        expect(component).toMatchSnapshot()
    });
    it('should contain the name "Miss Jazz"', () => {
        expect(wrapper.find('h1').text()).toEqual('Miss Jazz');
    });
    test('should contain a profile image', () => {
        expect(wrapper.find('img').prop('src')).toEqual('profilelogo.png');
    });
    it('should contain the job title "Software Engineer"', () => {
        expect(wrapper.find('.title').text()).toEqual('Software Engineer');
    });

    it('should contain the company name "Reckonsys Tech Labs"', () => {
        expect(wrapper.find('p').at(1).text()).toEqual('Reckonsys Tech Labs');
    });
    test('should contain a link to the dashboard', () => {
        const link = wrapper.find(Link);
        expect(link.prop('to')).toEqual('/dashboard');
        expect(link.text()).toEqual('Close');
    });

})
