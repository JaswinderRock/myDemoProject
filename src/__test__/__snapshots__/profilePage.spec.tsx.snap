// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Profile Page Profile test is render 1`] = `ShallowWrapper {}`;
